src/APIInterfaces/BackendAPI/RequestTypes/CourseRequestTypes.ts

```diff
@@ -55,7 +55,20 @@ export interface DeleteCourseTopicOptions {
 export interface ExtendCourseTopicForUser {
     courseTopicContentId: number;
     userId: number;
-    extensions: {startDate: Moment, endDate: Moment, deadDate: Moment};
+    topicAssessmentInfoId?: number;
+    data: {
+        extensions?: {
+            startDate?: Moment;
+            endDate?: Moment;
+            deadDate?: Moment;
+        };    
+        studentTopicAssessmentOverride?: {
+            versionDelay?: number;
+            duration?: number;
+            maxVersions?: number;
+            maxGradedAttemptsPerVersion?: number;
+        };
```

Comment:
would it ever be relevant for these sub-objects to have their own interfaces split out?

Reply:
It'd be nice once we have the refactor, but I don't think there's any short-term reason.

Reply:
I don't think so, eventually, this will be coming directly from the validation library